src/mscorlib/shared/System/Numerics/Vector.cs

```diff
@@ -2,9 +2,11 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using Internal.Runtime.CompilerServices;
```

Comment:
This needs to be under ifdef. Otherwise, it won't build once mirrored to CoreFX used to build the portable Vector version.

Reply:
This single instancing is in place already. Once this PR is merged, there will be PR with your change created in corefx automatically: https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/README.md

We use `#if !netstandard` in other similar places: https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Runtime/CompilerServices/ConfiguredValueTaskAwaitable.cs#L11 . I would try to stick with that unless if does not work for some reason.

Reply:
In CoreFx, we include this file when building for `netfx` and `netstandard`.  So I'm guessing `!netstandard` isn't going to be enough.  

Also - the whole new constructor should probably be under the `ifdef`, so it doesn't exist in the `netfx` and `netstandard` versions.

Reply:
@eerhardt I again went through the csproj for CoreFx project and in there Vector.cs is included if not targetting `netcoreapp`. I am thoroughly confused when it comes to nomenclature such as `netstandard` vs `netfx`, `netcoreapp`, etc. So do you mean to say that I should put the `using` statement and entire new constructor under `#if (!netfx && !netstandard)`.

I am not able to understand how this `#if` compares to the conditional compilation in the CoreFx csproj. Hence please help with what I should actually use.

Reply:
> I am thoroughly confused when it comes to nomenclature such as netstandard vs netfx, netcoreapp, etc.

`netcoreapp` means `.NET Core`.
`netfx` means `.NET Framework`.

Both the `.NET Core` and `.NET Framework` are concrete implementations of a runtime that executes .NET code. This means you can actually run code on them. The `.NET Framework` ships as part of Windows and only runs on Windows.  `.NET Core` runs on Windows, Linux and OSX.

You can think of `netstandard` as an abstraction a developer can target that will allow their code to run on any concrete runtime: `.NET Core`, `.NET Framework`, `Mono`, etc.  `netstandard` isn't a runtime itself, but it is a contract that the concrete runtimes implement.

> So do you mean to say that I should put the using statement and entire new constructor under #if (!netfx && !netstandard).

I think it should actually be under `#if CORECLR` (which is the [DefineConstant used by coreclr](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/System.Private.CoreLib.csproj#L36) to mean "this is `netcoreapp`").

When building this code in dotnet/corefx for `netfx` and `netstandard` that DefineConstant won't be defined, so this code won't be compiled.

Reply:
> When building this code in dotnet/corefx for netfx and netstandard that DefineConstant won't be defined, so this code won't be compiled.

Thanks @eerhardt for the explanation. Actually what confused is me that if `netstandard` is the standard and `netfx` is implementation then why was I supposed to use both in that `#if`. And just when I thought I had reconfirmed my understanding by reading your above reply, I again got stuck at the last line in which you have mentioned `netfx` and `netstandard` in same line :-) . I thought `netstandard` includes `netfx` (standard and implementation relationship). So did you want to mention `netcoreapp` instead of `netstandard` ...? In other words is it necessary to explicitly say `netfx` when I say `netstandard`... Hope I am not creating more confusion.

As for the PR, I am replacing it iwth `CORECLR` as you have mentioned.

Reply:
>  iwth CORECLR as you have mentioned.

That will not compile on CoreRT and ProjectN. It would need to be more combined with more conditions there. I think `#if netcoreapp` and defining netcoreapp when building S.P.CoreLib  as you have proposed originally is probably the best choice.

Reply:
@jkotas Thank you. I have pushed a commit that defines `netcoreapp` and uses in the .cs file.